app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt

```diff
@@ -559,7 +559,55 @@ class ChatController(args: Bundle) : BaseController(args), MessagesListAdapter
         }
     }
 
-    private fun showBrowserScreen(browserType: BrowserController.BrowserType) {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if(requestCode == REQUEST_CODE_CHOOSE_FILE){
+            if (resultCode == RESULT_OK) {
+                uploadFile(data)
+            }
+        }
+    }
+
+    private fun uploadFile(intentData: Intent?) {
+        try {
+            checkNotNull(intentData)
+            val files: MutableList<String> = ArrayList()
+            intentData.clipData?.let {
+                for (index in 0 until it.itemCount) {
+                    files.add(it.getItemAt(index).uri.toString())
+                }
+            } ?: run {
+                checkNotNull(intentData.data)
+                intentData.data.let {
+                    files.add(intentData.data.toString())
+                }
+            }
+            require(files.isNotEmpty())
+            val data: Data = Data.Builder()
+                    .putStringArray(UploadAndShareFilesWorker.DEVICE_SOURCEFILES, files.toTypedArray())
+                    .putString(UploadAndShareFilesWorker.NC_TARGETPATH, conversationUser?.getAttachmentFolder())
+                    .putString(UploadAndShareFilesWorker.ROOM_TOKEN, roomToken)
+                    .build()
+            val uploadWorker: OneTimeWorkRequest = OneTimeWorkRequest.Builder(UploadAndShareFilesWorker::class.java)
+                    .setInputData(data)
+                    .build()
+            WorkManager.getInstance().enqueue(uploadWorker)
+        } catch (e: IllegalStateException){
+            Toast.makeText(context, context?.resources?.getString(R.string.nc_upload_failed), Toast.LENGTH_LONG).show()
+            Log.e(javaClass.simpleName, "Something went wrong when trying to upload file", e)
+        } catch (e: IllegalArgumentException) {
+            Toast.makeText(context, context?.resources?.getString(R.string.nc_upload_failed), Toast.LENGTH_LONG).show()
+            Log.e(javaClass.simpleName, "Something went wrong when trying to upload file", e)
+        }
+    }
+
+    fun sendSelectLocalFileIntent() {
+        var action = Intent(Intent.ACTION_GET_CONTENT)
+        action = action.setType("*/*").addCategory(Intent.CATEGORY_OPENABLE)
+        action.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
+        startActivityForResult(Intent.createChooser(action, context?.resources?.getString(R.string.nc_upload_choose_local_files)), REQUEST_CODE_CHOOSE_FILE);
```

Comment:
For reviewing on GitHub best approach is to have max line length of 120 characters.
If line is too long, we put each argument in a new line